### DISCUSSION TITLE
Reduce the number of third party requests

### DISCUSSION THREAD

**Discussion by tobiasff3200** [home-assistant/frontend#8379] [Other feature requests] [answered]
While navigating through the frontend many requests to third party servers are sent:
- cloudflare.com (scripts and CSS)
- frenck.dev (cookie and images)
- github.com (cookies and image)
- evwilkin.github.io (image)
- avatarsX.githubusercontent.com (images)
- drive.google.com (images)
- fonts.googleapis.com (CSS)
- home-assistant.io (cookie)
- brands.home-assistant.io (images)
- img.shields.io (images)
- basemaps.cartocdn.io (images)

The reason why I host a smart home system myself is that I do not want that other know what I do and collect telemetry. My intention is not to blame anyone (I love home-assistant) but that many third party requests cause severe privacy issues and could easily be avoided by shipping the files as part of the software. This also means that home-assistant is not fully usable offline (e.g. the File Editor do not work at all without third party requests).

It would be great if these files (or at least most of them) are shipped with home-assistant and served from the local server.

**Comment by iMicknl** [comment, marked as answer]
@TobiasMie are you running HACS perhaps? On my Home Assistant installation all front-end bits are served locally, only the brand images come from `brands.homeassistant.io`. 

Looking at the other requests, it looks like your are using HACS or you are opening addon pages via the Supervisor. For both, it will show the README from GitHub, which can include external sources. I believe that most of your extra requests are coming from these pages.

**Reply by tobiasff3200** [reply]
Yes I am running HACS but that wasn't the problem.
I identified the [File Editor](https://github.com/home-assistant/addons/tree/master/configurator) as the source of all these requests.
Thanks for replying, I will open an issue in this repository.

**Reply by iMicknl** [reply]
@TobiasMie good to know! The following requests could come from HACS + Supervisor as well;
frenck.dev (cookie and images)
github.com (cookies and image)
evwilkin.github.io (image)
avatarsX.githubusercontent.com (images)
img.shields.io (images)
